src/repeated_game.jl

```diff
@@ -438,3 +440,118 @@ function outerapproximation(rpd::RepGame2; nH=32, tol=1e-8, maxiter=500,
 
     return vertices
 end
+
+
+function AS{T}(rpd::RepeatedGame{2, T}; max_iter::Integer=1000,
+               plib=getlibraryfor(2, T), tol::Float64=1e-5, u=nothing)
+
+    # Initialize W0 with each entries of payoff bimatrix
+    v_old = _payoff_points(rpd.sg)
+
+    if u == nothing
+        u = [minimum(rpd.sg.players[1].payoff_array),
+             minimum(rpd.sg.players[2].payoff_array)]
+    end
+
+    # create VRepresentation and Polyhedron
+    V = SimpleVRepresentation(v_old)
+    p = polyhedron(V, plib)
+    H = SimpleHRepresentation(p)
+    # get rid of redundant vertices
+    V = removevredundancy(V, H)
+    p = polyhedron(V, plib)
+
+    # calculate the best deviation gains
+    # normalize with (1-delta)/delta
+    best_dev_gains1, best_dev_gains2 = (1-rpd.delta)/rpd.delta .* _best_dev_gains(rpd.sg)
+
+    for iter = 1:max_iter
+
+        v_new = Vector{T}(0) # to store new vertices
+        # step 1
+        for a2 in 1:rpd.sg.nums_actions[2]
+            for a1 in 1:rpd.sg.nums_actions[1]
+                payoff1 = rpd.sg.players[1].payoff_array[a1, a2]
+                payoff2 = rpd.sg.players[2].payoff_array[a2, a1]
+                IC1 = u[1] + best_dev_gains1[a1, a2]
+                IC2 = u[2] + best_dev_gains2[a2, a1]
+
+                # check if the payoff point is interior
+                # first check if it satisifies IC
+                if all([payoff1, payoff2] .> [IC1, IC2])
+                    # then check if it is in the polyhedron
+                    if [payoff1, payoff2] in H
+                        push!(v_new, payoff1, payoff2)
+                    end
+                end
+
+                # find out the intersections of polyhedron and IC boundaries
+                p_IC = polyhedron(SimpleHRepresentation(-eye(2), -[IC1, IC2]), plib)
```

Comment:
IC constraints will change when `u` is updated. But yeah, I think there is no need to update `p_IC` in every iteration. I will move this into [here](https://github.com/shizejin/Games.jl/blob/9c718205b5041571bc6090b866590704bc2bd491/src/repeated_game.jl#L530) so that `p_IC` is updated only if `u` changes. This will save a lot of time! Thanks.